### PR TITLE
Add SGF endpoint

### PR DESCRIPTION
**Pull request by rchs819**
Enhancement for #27  
Added SGF endpoint `/viewmatch/<sgf_hash>.sgf`
Added SGF download link on Match Game page

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1339,7 +1339,8 @@ app.get('/match-games/:matchid(\\w+)', (req, res) => {
                 html += "<td>" + ipMap.get(item.ip) + "</td>";
                 html += "<td><a href=\"/viewmatch/" + item.sgfhash + "?viewer=wgo\">" + item.sgfhash + "</a></td>";
                 html += "<td>" + item.winnerhash.slice(0,8) + "</td>";
-                html += "<td>" + item.score + "</td><td>" + item.movescount + "</td></tr>\n";
+                html += "<td>" + item.score + "</td><td>" + item.movescount + "</td>";
+				html += "<td><a href=\"/viewmatch/" + item.sgfhash + ".sgf\">sgf</a></td></tr>\n";
```
Comment (roy7) [inline server.js]: Can you format this with spaces instead of tabs?

File: server.js [hunk 2]
```diff
@@ -1354,6 +1355,24 @@ app.get('/match-games/:matchid(\\w+)', (req, res) => {
 
 });
 
+app.get('/viewmatch/:hash(\\w+)\\.sgf', (req, res) => {
+	Promise.all([
```
Comment (roy7) [inline server.js]: Spaces instead of tab please :)

**Comment by roy7** [comment]
Awesome! I added a couple minor comments to code.

File: server.js (hunk 1 above)
Comment (rchs819) [reply server.js]: Sorry for the tabs, I'll adjust my IDE to use space instead of tabs.

**Comment by roy7** [comment]
Hmm merged it, but I get

Cannot GET /viewmatch/194063ae93c513e1c4425d2dc096cc65eadb5d450cf00c44c3d64d862d138757.sgf

When clicking on an SGF link. Will see what I can figure out.

**Comment by roy7** [comment]
No it's a node level error like it can't find a valid handler.

http://zero.sjeng.org/viewmatch/c4329750e2601eba5a6ccd5d0c00b05d815385fb3e972422852fc6cee5d3e933.sgf

**Comment by roy7** [comment]
Fixed it. The period didn't need escaped. It's a literal character in the regex-to-path node stuff.

**Comment by rchs819** [comment]
Cool, the downloaded sgf looks good on my side.

**Pull request merged**